pkg/volume/glusterfs/glusterfs.go

```diff
@@ -492,9 +513,43 @@ func (r *glusterfsVolumeProvisioner) Provision() (*v1.PersistentVolume, error) {
 		return nil, err
 	}
 	r.provisioningConfig = *cfg
-
 	glog.V(4).Infof("glusterfs: creating volume with configuration %+v", r.provisioningConfig)
-	glusterfs, sizeGB, err := r.CreateVolume()
+	scName := r.options.PVC.Annotations["volume.beta.kubernetes.io/storage-class"]
+	
+	// TODO :Release GIDs
+	if startGid64, err := strconv.ParseUint(r.provisioningConfig.gidMin, 10, 32); err == nil {
+		blockStart32 = uint32(startGid64)
+		glog.V(4).Infof("glusterfs: blockstart details %#v", blockStart32)
+	}
+	if endGid64, err := strconv.ParseUint(r.provisioningConfig.gidMax, 10, 32); err == nil {
+		blockEnd32 = uint32(endGid64)
+		glog.V(4).Infof("glusterfs: blockend details %#v ", blockEnd32)
+	}
+	glog.V(2).Infof("glusterfs: provisioning config and gid range %#v [%#v - %#v]", r.provisioningConfig, blockStart32, blockEnd32)
+	gidAllocatorp := r.plugin.gidallocatormap[scName]
+	if scName != "" {
+		if gidAllocatorp != nil {
+			glog.V(1).Infof("glusterfs: bitmap : %#v exist for %s ", gidAllocatorp, scName)
+		} else {
+			gidAllocPtr, err := gidPoolCreate(blockStart32, blockEnd32)
```

Comment:
I see new allocator created here, but I do not see where it is stored into `d.plugin.gidallocatormap`. Do I miss something?